### ISSUE TITLE
Provider produced invalid plan

### ISSUE THREAD

**Issue by mrarar** [cloudflare/terraform-provider-cloudflare#2258]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

terraform: 1.3.9
cloudflare: 4.0.0

### Affected resource(s)

efs_volumes

### Terraform configuration files

```hcl
main.tf

provider "cloudflare" {
  email   = var.cloudflare_email
  api_key = var.cloudflare_api_key
}
```

### Link to debug output

https://gist.github.com/mrarar/0626767498350b8049f4e06a8327c599

### Expected output

Successful terraform plan. 

### Actual output

│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_record.efs_volumes_stg.allow_overwrite: planned value
│ cty.False for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_record.efs_volumes_stg.tags: planned value
│ cty.SetValEmpty(cty.String) for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_record.efs_volumes_stg.comment: planned value
│ cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_record.efs_volumes_stg.proxied: planned value
│ cty.False for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.

### Steps to reproduce

terraform init.
terraform plan.

**Comment by jacobbednarz** [comment]
can you please provide a minimal reproduction case for this issue? the configuration you have provided doesn't have the resource that is related to this issue but plenty of unrelated setup.

**Comment by jacobbednarz** [comment]
plugging in a static value for the record `value` works fine here.

**Comment by mrarar** [comment]
It appears what is happening is that the cloudflare_record resource is forcing the optional configs and would not pass my terraform plan until I added in values to `allow_overwrite`, `comment`, `proxied`, and `tags`.

**Comment by mrarar** [comment]
It appears what is happening is that the cloudflare_record resource is forcing the optional configs and would not pass my terraform plan until I added in values to `allow_overwrite`, `comment`, `proxied`, and `tags`.

**Comment by jacobbednarz** [comment]
i'm not seeing that behaviour here. it's still using sdkv2 so it doesn't require explicit values. have you modified your state file manually at any point?

**Comment by saikrishna434** [comment]
same even for us

`[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[2].tags: planned value
[31m│[0m [0mcty.SetValEmpty(cty.String) for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[2].allow_overwrite: planned value
[31m│[0m [0mcty.False for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[2].proxied: planned value cty.False for a
[31m│[0m [0mnon-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[2].comment: planned value
[31m│[0m [0mcty.StringVal("") for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[3].allow_overwrite: planned value
[31m│[0m [0mcty.False for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[3].tags: planned value
[31m│[0m [0mcty.SetValEmpty(cty.String) for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[3].proxied: planned value cty.False for a
[31m│[0m [0mnon-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[3].comment: planned value
[31m│[0m [0mcty.StringVal("") for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[0].allow_overwrite: planned value
[31m│[0m [0mcty.False for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[0].tags: planned value
[31m│[0m [0mcty.SetValEmpty(cty.String) for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[0].proxied: planned value cty.False for a
[31m│[0m [0mnon-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[0].comment: planned value
[31m│[0m [0mcty.StringVal("") for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[1].tags: planned value
[31m│[0m [0mcty.SetValEmpty(cty.String) for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[1].allow_overwrite: planned value
[31m│[0m [0mcty.False for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[1].proxied: planned value cty.False for a
[31m│[0m [0mnon-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m
[31m╷[0m[0m
[31m│[0m [0m[1m[31mError: [0m[0m[1mProvider produced invalid plan[0m
[31m│[0m [0m
[31m│[0m [0m[0mProvider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
[31m│[0m [0mvalue for cloudflare_record.main[1].comment: planned value
[31m│[0m [0mcty.StringVal("") for a non-computed attribute.
[31m│[0m [0m
[31m│[0m [0mThis is a bug in the provider, which should be reported in the provider's
[31m│[0m [0mown issue tracker.
[31m╵[0m[0m`

**Comment by zeon256** [comment]
> It appears what is happening is that the cloudflare_record resource is forcing the optional configs and would not pass my terraform plan until I added in values to `allow_overwrite`, `comment`, `proxied`, and `tags`.

Cloudflare: 4.0.0

I can confirm that adding these values worked with the exception of tags. If I call `apply`, I have to comment `tags` but calling destroy without `tags` gave `Error: Provider produced invalid plan`
```hcl
resource "cloudflare_record" "associate_domain" {
  zone_id         = var.zone_id
  name            = var.sub_domain
  value           = var.my_ipv4_address
  type            = "A"
  ttl             = 60
  allow_overwrite = true
  comment         = "some comment"
  tags            = [""] # dont add this line when apply if not will error 1004, but add back this line when destroying
  proxied = false
}
```

**Comment by edumgui** [comment]
I was doing a POC and I had to use version 3.31.0 of the provider for it to run properly without `comments` and `tags` arguments. The issue starts on 3.32.0 https://github.com/cloudflare/terraform-provider-cloudflare/commit/3401db071cea511b73e9347095dff3455b512be4

Also setting `tags` will not work if you are using a free account: https://developers.cloudflare.com/dns/manage-dns-records/reference/record-attributes/#record-tags

**Comment by poliva-uturn** [comment]
This worked for me as well @edumgui

**Comment by wilson-urdaneta** [comment]
Same here, we have to pin back to 3.32 in our case

**Comment by sacedo-asapp** [comment]
Something similar is happening to me but with the `cloudflare_access_identity_provider` resource. This was working well in version 3. 

```
 Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].apps_domain:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].centrify_app_id: planned
│ value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].onelogin_account:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].certs_url:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].sign_request:
│ planned value cty.False for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].token_url:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].directory_id:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].auth_url:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].centrify_account:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].email_attribute_name:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].idp_public_cert: planned
│ value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for cloudflare_access_identity_provider.okta.config[0].issuer_url:
│ planned value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].sso_target_url: planned
│ value cty.StringVal("") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for
│ cloudflare_access_identity_provider.okta.config[0].support_groups: planned
│ value cty.False for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
Cleaning up project directory and file based variables
00:01
ERROR: Job failed: command terminated with exit code 1
```

**Comment by triwats** [comment]
We're seeing this too, tried the approach of providing an empty array of tags as suggested by @zeon256 but alas when trying to create a record:

``` hcl
╷
│ Error: failed to create DNS record: DNS record tag name must not be empty. (9322)
│ 
│   with cloudflare_record.auth_cname,
│   on cloudflare.tf line 1, in resource "cloudflare_record" "foo_cname":
│    1: resource "cloudflare_record" "foo_cname" 
│ 
```

Cloudflare Provider Version: v4.5.0

It's also good to know that the provider wants it as a `Set of Strings` as a type. Look at the wider [API reference](https://developers.cloudflare.com/dns/manage-dns-records/reference/record-attributes/#tags), it's wanted as `key:value`.

**Comment by tuan-phan** [comment]
Same issue with different resources:

```
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for module.web["test-news.bitcoin.com"].cloudflare_zone.zone.paused:
│ planned value cty.False for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵
╷
│ Error: Provider produced invalid plan
│ 
│ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid
│ value for module.web["test-news.bitcoin.com"].cloudflare_zone.zone.type:
│ planned value cty.StringVal("full") for a non-computed attribute.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
╵

```

**Comment by jacobbednarz** [comment]
thanks @simpson-ross 🙇 that helps clarify things here.

> TF version 1.1.4 and prior exhibit this error

this is expected since terraform < 1.1.5 had an issue (hashicorp/terraform#30330) which incorrectly marked fields as `nullable=false`. it was already addressed in 1.2.x but this fix was backported to 1.1.5 if you must run the 1.x release series. 

```
$ ASDF_TERRAFORM_VERSION=1.1.5 terraform apply
╷
│ Error: Unsupported Terraform Core version
│ 
│   on terraform.tf line 40, in terraform:
│   40:   required_version = "1.1.4"
│ 
│ This configuration does not support Terraform version 1.1.5. To proceed, either choose another supported Terraform version or update this version constraint. Version constraints are normally set for good reason, so updating the constraint may lead to other errors or unexpected
│ behavior.
```

```
$ ASDF_TERRAFORM_VERSION=1.1.5 terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # cloudflare_record.mx_1 will be created
  + resource "cloudflare_record" "mx_1" {
      + allow_overwrite = false
      + created_on      = (known after apply)
      + hostname        = (known after apply)
      + id              = (known after apply)
      + metadata        = (known after apply)
      + modified_on     = (known after apply)
      + name            = "@"
      + priority        = 5
      + proxiable       = (known after apply)
      + ttl             = 300
      + type            = "MX"
      + value           = "route1.mx.cloudflare.net"
      + zone_id         = "0da42c8d2132a9ddaf714f9e7c920711"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

cloudflare_record.mx_1: Creating...
cloudflare_record.mx_1: Creation complete after 0s [id=318a0b16eea91338ef42a36ef8d51adc]
```

do you see this behaviour with terraform > 1.2 and the 4.x release of the provider?

**Comment by OzoneNZ** [comment]
> thanks @simpson-ross 🙇 that helps clarify things here.
> 
> > TF version 1.1.4 and prior exhibit this error
> 
> this is expected since terraform < 1.1.5 had an issue ([hashicorp/terraform#30330](https://github.com/hashicorp/terraform/pull/30330)) which incorrectly marked fields as `nullable=false`. it was already addressed in 1.2.x but this fix was backported to 1.1.5 if you must run the 1.x release series.
> 
> ```
> $ ASDF_TERRAFORM_VERSION=1.1.4 terraform apply
> ╷
> │ Error: Provider produced invalid plan
> │ 
> │ Provider "registry.terraform.io/cloudflare/cloudflare" planned an invalid value for cloudflare_record.mx_1.allow_overwrite: planned value cty.False for a non-computed attribute.
> │ 
> │ This is a bug in the provider, which should be reported in the provider's own issue tracker.
> ╵
> ```
> 
> ```
> $ ASDF_TERRAFORM_VERSION=1.1.5 terraform apply
> 
> Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
>   + create
> 
> Terraform will perform the following actions:
> 
>   # cloudflare_record.mx_1 will be created
>   + resource "cloudflare_record" "mx_1" {
>       + allow_overwrite = false
>       + created_on      = (known after apply)
>       + hostname        = (known after apply)
>       + id              = (known after apply)
>       + metadata        = (known after apply)
>       + modified_on     = (known after apply)
>       + name            = "@"
>       + priority        = 5
>       + proxiable       = (known after apply)
>       + ttl             = 300
>       + type            = "MX"
>       + value           = "route1.mx.cloudflare.net"
>       + zone_id         = "0da42c8d2132a9ddaf714f9e7c920711"
>     }
> 
> Plan: 1 to add, 0 to change, 0 to destroy.
> 
> Do you want to perform these actions?
>   Terraform will perform the actions described above.
>   Only 'yes' will be accepted to approve.
> 
>   Enter a value: yes
> 
> cloudflare_record.mx_1: Creating...
> cloudflare_record.mx_1: Creation complete after 0s [id=318a0b16eea91338ef42a36ef8d51adc]
> ```
> 
> do you see this behaviour with terraform > 1.2 and the 4.x release of the provider?

Can confirm after testing many different Terraform versions last week I found that Terraform versions >= 1.1.5 didn't exhibit this bug - nice to know for sure it was a Terraform bug, cheers!

**Comment by jacobbednarz** [comment]
there is a note in the README now that states we expect Terraform > 1.2 which should cover everything we've raised here. if you're still seeing issues with another combination, please open a dedicated issue to that following the issue template.
